pkg/processor/runtime/rpc/connection/connectionallocator.go

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright 2025 The Nuclio Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package connection
+
+import (
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/nuclio/errors"
+)
+
+type ConnectionAllocator struct {
+	*AbstractConnectionManager
+
+	serverAddress string
+
+	// should be a buffered chan when support multiple
+	eventConnections []*Connection
+}
+
+func NewConnectionAllocator(abstractConnectionManager *AbstractConnectionManager) *ConnectionAllocator {
+	return &ConnectionAllocator{
+		AbstractConnectionManager: abstractConnectionManager,
+		serverAddress: fmt.Sprintf("%s:%d",
+			abstractConnectionManager.Configuration.host,
+			abstractConnectionManager.Configuration.port),
+		eventConnections: make([]*Connection, 0),
+	}
+}
+
+func (ca *ConnectionAllocator) Prepare() error {
+	if err := ca.prepareControlMessageSocket(); err != nil {
+		return errors.Wrap(err, "Failed to prepare control message socket")
+	}
+	return nil
+}
+
+func (ca *ConnectionAllocator) Start() error {
+	if ca.MinConnectionsNum != 0 {
+		for i := 0; i < ca.MinConnectionsNum; i++ {
+			var conn net.Conn
+			var err error
+
+			// Use retryable dial for the first connection
+			if i == 0 {
+				conn, err = retryableDial(ca.serverAddress, 30, 1*time.Second)
+			} else {
+				conn, err = net.Dial("tcp", ca.serverAddress)
+			}
```

Comment:
why not always use a retryable dial?

Reply:
@TomerShor because it may fail here only for case when wrapper isn't fully started yet. Otherwise, something is off